pkg/kubelet/pleg/generic.go

```diff
@@ -122,34 +165,66 @@ func (g *GenericPLEG) relist() {
 		return
 	}
 
+	// Build a new map.
+	podMap := buildPodMap(pods)
```

Comment:
Added a comment.

I am a bit concerned about this too, but not sure if it's worth optimizing. I can also keep the list of pods and just iterate through them if needed.